lib/action_subscriber.rb

```diff
@@ -62,8 +60,10 @@ def self.print_subscriptions
   end
 
   def self.setup_queues!
-    ::ActionSubscriber::Base.inherited_classes.each do |klass|
-      klass.setup_queues!
+    @route_set ||= begin
```

Comment:
If you split initializing the route set out from the setup queues step, you won't need to add any guard clauses. I'd follow the same pattern as initializing the config in this class.